image/directory/directory_src.go

```diff
@@ -26,7 +26,25 @@ type dirImageSource struct {
 
 // newImageSource returns an ImageSource reading from an existing directory.
 // The caller must call .Close() on the returned ImageSource.
-func newImageSource(ref dirReference) private.ImageSource {
+func newImageSource(ref dirReference) (private.ImageSource, error) {
+	versionPath := ref.versionPath()
+	contents, err := os.ReadFile(versionPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, fmt.Errorf("version file not found at %q: not a valid directory transport image", versionPath)
```

Comment:
I’m worried this might break some users who create `dir:` manually. (I have no data to support or refute that.)

Reply:
removed this in favour of assuming 1.1 if no version file.